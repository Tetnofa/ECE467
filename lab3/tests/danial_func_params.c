int main(int a) {
    a = 1;
}

int fcn(int f) {
    int x;
    f = x;
}


int fcn1(int f) {
    int a;
    a = f;
    fcn(f);
}
void fcn2(bool x, bool z, bool k) {
    int y;
    y = x;
    x = z;
    k = z;
    f = 1;
}
int fcn4(int pee[]);
int fcn4(int pee[]) {
    fcn4(pee);
}
void fcn3(int arr[]) {
    fcn2(arr[2]);
    fcn4(arr);
}

int pee(int a, int b, int c);
int pee(int a, int b, int c) {
    int arr[2];
    arr[1] = a;
    arr[2] = b;
    arr[3] = c;
}