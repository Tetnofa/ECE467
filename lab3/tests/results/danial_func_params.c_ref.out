Program [useIO=0](1,0)
	Function[isProto=0]	Int(1,0)
		Identifier[name:main](1,4)
		Parameter		Int(1,9)
			Identifier[name:a](1,13)
		Scope(1,16)
			Assignment(2,4)
				Reference(2,4)
					Identifier[name:a](2,4)
				Int Expression(2,8)
					IntConstant[val=1](2,8)
	Function[isProto=0]	Int(5,0)
		Identifier[name:fcn](5,4)
		Parameter		Int(5,8)
			Identifier[name:f](5,12)
		Scope(5,15)
			Scalar Declaration			Int(6,4)
				Identifier[name:x](6,8)
			Assignment(7,4)
				Reference(7,4)
					Identifier[name:f](7,4)
				Int Expression(7,8)
					Reference(7,8)
						Identifier[name:x](7,8)
	Function[isProto=0]	Int(11,0)
		Identifier[name:fcn1](11,4)
		Parameter		Int(11,9)
			Identifier[name:f](11,13)
		Scope(11,16)
			Scalar Declaration			Int(12,4)
				Identifier[name:a](12,8)
			Assignment(13,4)
				Reference(13,4)
					Identifier[name:a](13,4)
				Int Expression(13,8)
					Reference(13,8)
						Identifier[name:f](13,8)
			Expression Statement(14,4)
				CallExpr(14,4)
					Identifier[name:fcn](14,4)
					Argument(14,8)
						Int Expression(14,8)
							Reference(14,8)
								Identifier[name:f](14,8)
	Function[isProto=0]	Void(16,0)
		Identifier[name:fcn2](16,5)
		Parameter		Bool(16,10)
			Identifier[name:x](16,15)
		Parameter		Bool(16,18)
			Identifier[name:z](16,23)
		Parameter		Bool(16,26)
			Identifier[name:k](16,31)
		Scope(16,34)
			Scalar Declaration			Int(17,4)
				Identifier[name:y](17,8)
			Assignment(18,4)
				Reference(18,4)
					Identifier[name:y](18,4)
				Int Expression(18,8)
					Reference(18,8)
						Identifier[name:x](18,8)
			Assignment(19,4)
				Reference(19,4)
					Identifier[name:x](19,4)
				Int Expression(19,8)
					Reference(19,8)
						Identifier[name:z](19,8)
			Assignment(20,4)
				Reference(20,4)
					Identifier[name:k](20,4)
				Int Expression(20,8)
					Reference(20,8)
						Identifier[name:z](20,8)
			Assignment(21,4)
				Reference(21,4)
					Identifier[name:f](21,4)
				Int Expression(21,8)
					IntConstant[val=1](21,8)
	Function[isProto=1]	Int(23,0)
		Identifier[name:fcn4](23,4)
		Parameter		Int*(23,9)
			Identifier[name:pee](23,13)
	Function[isProto=0]	Int(24,0)
		Identifier[name:fcn4](24,4)
		Parameter		Int*(24,9)
			Identifier[name:pee](24,13)
		Scope(24,20)
			Expression Statement(25,4)
				CallExpr(25,4)
					Identifier[name:fcn4](25,4)
					Argument(25,9)
						Int Expression(25,9)
							Reference(25,9)
								Identifier[name:pee](25,9)
	Function[isProto=0]	Void(27,0)
		Identifier[name:fcn3](27,5)
		Parameter		Int*(27,10)
			Identifier[name:arr](27,14)
		Scope(27,21)
			Expression Statement(28,4)
				CallExpr(28,4)
					Identifier[name:fcn2](28,4)
					Argument(28,9)
						Int Expression(28,9)
							Reference(28,9)
								Identifier[name:arr](28,9)
								Int Expression(28,13)
									IntConstant[val=2](28,13)
			Expression Statement(29,4)
				CallExpr(29,4)
					Identifier[name:fcn4](29,4)
					Argument(29,9)
						Int Expression(29,9)
							Reference(29,9)
								Identifier[name:arr](29,9)
	Function[isProto=1]	Int(32,0)
		Identifier[name:pee](32,4)
		Parameter		Int(32,8)
			Identifier[name:a](32,12)
		Parameter		Int(32,15)
			Identifier[name:b](32,19)
		Parameter		Int(32,22)
			Identifier[name:c](32,26)
	Function[isProto=0]	Int(33,0)
		Identifier[name:pee](33,4)
		Parameter		Int(33,8)
			Identifier[name:a](33,12)
		Parameter		Int(33,15)
			Identifier[name:b](33,19)
		Parameter		Int(33,22)
			Identifier[name:c](33,26)
		Scope(33,29)
			Array Declaration			Int[2](34,4)
				Identifier[name:arr](34,8)
			Assignment(35,4)
				Reference(35,4)
					Identifier[name:arr](35,4)
					Int Expression(35,8)
						IntConstant[val=1](35,8)
				Int Expression(35,13)
					Reference(35,13)
						Identifier[name:a](35,13)
			Assignment(36,4)
				Reference(36,4)
					Identifier[name:arr](36,4)
					Int Expression(36,8)
						IntConstant[val=2](36,8)
				Int Expression(36,13)
					Reference(36,13)
						Identifier[name:b](36,13)
			Assignment(37,4)
				Reference(37,4)
					Identifier[name:arr](37,4)
					Int Expression(37,8)
						IntConstant[val=3](37,8)
				Int Expression(37,13)
					Reference(37,13)
						Identifier[name:c](37,13)
sema: 18:4 : type mismatch 
sema: 21:4 : use of undefined identifier f
sema: 21:4 : type mismatch 
sema: 28:4 : no matching definition for fcn2
